cli/deno_dir.rs

```diff
@@ -1064,64 +1035,127 @@ mod tests {
+        fs::read_to_string(&target_path).unwrap(),
         "export const redirect = 1;\n"
       );
       let redirect_target_headers =
-        deno_dir.get_source_code_headers(&target_module_url);
+        deno_dir.get_source_code_headers(&target_url);
       assert!(redirect_target_headers.redirect_to.is_none());
 
       // Examine the meta result.
-      assert_eq!(mod_meta.url.clone(), target_module_url);
-      assert_eq!(
-        &mod_meta.redirect_source_url.clone().unwrap(),
-        &redirect_module_url
-      );
+      assert_eq!(mod_meta.url.clone(), target_url);
+    });
+  }
+
+  #[test]
+  fn test_get_source_code_5() {
+    let (_temp_dir, deno_dir) = test_setup();
+    // Test that redirect target is not downloaded twice for different redirect source.
+    tokio_util::init(|| {
+      let double_redirect_url =
+        Url::parse("http://localhost:4548/tests/subdir/redirects/redirect1.js")
+          .unwrap();
+
+      let redirect_url =
+        Url::parse("http://localhost:4546/tests/subdir/redirects/redirect1.js")
+          .unwrap();
+
+      let target_path = deno_dir
+        .deps_cache
+        .location
+        .join("http/localhost_PORT4545/tests/subdir/redirects/redirect1.js");
+
+      deno_dir
+        .get_source_file(&double_redirect_url, true, false)
+        .unwrap();
+
+      let result = fs::File::open(&target_path);
+      assert!(result.is_ok());
+      let file = result.unwrap();
+      // save modified timestamp for headers file of redirect target
+      let file_metadata = file.metadata().unwrap();
+      let file_modified = file_metadata.modified().unwrap();
+
+      // When another file is fetched that also point to redirect target, then redirect target
+      // shouldn't be downloaded again. It can be verified using source header file creation
+      // timestamp (should be the same as after first `get_source_file`)
+      deno_dir
+        .get_source_file(&redirect_url, true, false)
+        .unwrap();
+
+      let result = fs::File::open(&target_path);
+      assert!(result.is_ok());
+      let file_2 = result.unwrap();
+      // save modified timestamp for headers file
+      let file_metadata_2 = file_2.metadata().unwrap();
+      let file_modified_2 = file_metadata_2.modified().unwrap();
+
+      assert_eq!(file_modified, file_modified_2);
+    });
+  }
+
+  #[test]
+  fn test_get_source_code_6() {
+    let (_temp_dir, deno_dir) = test_setup();
+    // Test that redirections can be limited
+    tokio_util::init(|| {
+      let double_redirect_url =
+        Url::parse("http://localhost:4548/tests/subdir/redirects/redirect1.js")
+          .unwrap();
+
+      let result =
+        deno_dir.fetch_remote_source(&double_redirect_url, false, false, 2);
+      assert!(result.is_ok());
+      let result =
+        deno_dir.fetch_remote_source(&double_redirect_url, false, false, 1);
+      assert!(result.is_err());
+
+      // TODO: add better tests
```

Comment:
This isn't an actionable TODO. Be more specific.

Reply:
Ooops, missed that one. Fixed